### ISSUE TITLE
Sensors should use sensible units/quantities as output rather than just real

### ISSUE THREAD

**Issue by modelica-trac-importer** [modelica/Modelica#876]
**Reported by dietmarw on 15 Oct 2012 08:33 UTC**
From #872:
 * Majority of sensor outputs is unitless. See e.g.
   * `Modelica.Mechanics.Rotational.Sensors`
   * `Modelica.Mechanics.Translational.Sensors`
   * `Modelica.Thermal.HeatTransfer.Sensors`

To be consistent it should be defined the way it is done for example in  the sensors of `Modelica.Mechanics.MultiBody.Sensors` with both _final_ quantity and unit modification, e.g.
```
 Blocks.Interfaces.RealOutput r[3](each final quantity="Position",
                                   each final unit = "m")
```

----

Migrated-From: https://trac.modelica.org/Modelica/ticket/876

**Comment by modelica-trac-importer** [comment]
**Comment by anonymous on 15 Oct 2012 08:40 UTC**
Fixing this also fixes #830.

**Comment by modelica-trac-importer** [comment]
**Comment by anonymous on 2 Nov 2012 09:33 UTC**
[attachment:T876.diff These] are my proposed changes. Please cross check them for
 * `ComplexOutput` in `Modelica.Magnetic.FundamentalWave`
 * `RealInput` in `Modelica.Electrical.Analog.Basic`
 * unit in `Modelica.Thermal.HeatTransfer.Celsius.TemperatureSensor` (see #830)
It also fixes some comments that were missing.

**Comment by modelica-trac-importer** [comment]
**Comment by anonymous on 29 Nov 2012 12:43 UTC**
Any chance to see it fixed?

**Comment by modelica-trac-importer** [comment]
**Comment by dietmarw on 3 Dec 2012 15:48 UTC**
OK patch applied in f2f3947370f7cfdee9c0a0c20a1bddd0b626247e. Sorry for the delay but I waited for the library officer to look at this first (which he hasn't done yet). Applying anyway.

**Comment by modelica-trac-importer** [comment]
**Comment by anonymous on 4 Dec 2012 11:02 UTC**
Thanks Dietmar. You might also close #830 then.

**Comment by modelica-trac-importer** [comment]
**Comment by hansolsson on 12 Dec 2012 11:18 UTC**
I am not sure adding them in this way is a good idea, and really think the library officer should look at this.

Yes; the sensor-outputs should have unit and quanity, but it is already trivial to deduce that from their definition in the sensor-models, **except** for temperatures as seen in #830. I can understand that having the type on the sensor might help if there are unit-errors since it might be easier to find the unit-incorrect equation.


However, we already have Modelica.SIunits containing this information and if we actually want to add it we should investigate a way to reuse that information. 

Basically combine RealInput and Modelica.SIunits.Ampere in some way instead of 
```
  Modelica.Blocks.Interfaces.RealOutput i(final quantity="ElectricCurrent", 
    final unit="A") 
```
Repeating quantity and unit leads to issues such as #875, and we removed similar use of unit in #873 and we should not just add similar ones here.

**Comment by modelica-trac-importer** [comment]
**Comment by dietmarw on 12 Dec 2012 11:41 UTC**
I agree that the fix as applied here is basically a hack that is not ideal. So I'm setting this now to a blocker since the library officer will have to decide if
 1. the (hack-)fix should be left as it is
 2. be removed again
 3. find another possibility that uses already defined SIunits without the need to create new output or input types

**Comment by modelica-trac-importer** [comment]
**Comment by otter on 12 Dec 2012 14:15 UTC**
There seems to be only the following choices that both are not good:

 1. Do not have units, and rely on the tool that it deduces the units by unit propagation (but one can then also argue, to use this approach at other places and only define units where it is absolutely essential). Furthermore, even if a tool does not support unit propagation, it would be able to make some tests, if the unit is in the input/output of physical components.
 1. Define the units at the outputs, but then unnecessary copy operations take place (with the danger of introducing errors).

Unless there is a better proposal we should use (2) now (and might improve in the future, if someone has a good idea to formulate this in Modelica).

Therefore, I am closing this ticket (please, only re-open, if you have another alternative to define the unit).
